### DISCUSSION TITLE
Feature Request - simple import/export of 2D files

### DISCUSSION THREAD

**Discussion by YarlBoro** [guillaumechereau/goxel#357] [Ideas]
First part of this request is about import. I hope there could be implemented some way of importing 2D file as a voxel layer with 1 voxel of thickness. Please consider exposing this option without using reference image feature.

Second part, export. At the moment Goxel export features are optimized towards creating 3D models or creating artful graphics. There is no way - to my knowledge - to quickly export image of certain orthogonal view (TOP/BOTTOM/LEFT/RIGHT...etc) with image resolution matching exactly resolution of a particular model. Idea is, to get 100% sharp 2D file that matches resolution of voxel model in certain view. Trying to set camera that way is extremely time consuming, unprecize & in consequence unusable.

The gist of it, quick way to get 2D images in and out of Goxel like they are pixelart assets would be a boost in lots of scenarios. I'm absolutely sure that not only in game dev context. I hope that discription makes sense.

EDIT: Hell, maybe it would be even possible to import iOS clipboard image content as a voxel layer. One can dream, right? :)

**Comment by guillaumechereau** [comment]
For the import, this is possible by importing your image as an 'image layer', and then converting it to a mesh in the layer option.  Thinking of it, it would be nice to be able to apply a 2d image color as a sort of projected paint layer.

Agreed about the export with pixel perfect size.  I'll check how hard this is to implement.

**Comment by YarlBoro** [comment]
> For the import, this is possible by importing your image as an 'image layer'
I know how to do that on the desktop, but I can't do that in iPadOS because of UI differences.
It is probably hard to implement touch interface alternative for desktop top bar. Maybe on touch devices menu bar could be made into another panel? Just a thought

Color projection of image to voxels - Big thing. This opens so many venues when one can shape model graphics with voxel editor+2D editor hand in hand. I use this workflow almost daily (Qubicle^^), so I'm sure of its usefulness :)
